DrinkingPub/Pricelist.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Vsite.Oom.DrinkingPub
+{
+    internal class Pricelist
+    {
+        string Name;
+        public Dictionary<string, double> Items;
+
+        public Pricelist(string name) {
+            Name = name;
+            Items = new Dictionary<string, double>();
+        }
+
+        public void AddItem(string ItemName, double Price)
+        {
+            if (!Items.TryAdd(ItemName,Price))
```

Comment:
> Unlike the `Dictionary` indexer, `TryAdd` doesn't override the element if the element with the given key exists in the dictionary. If the key already exists, `TryAdd` does nothing and returns `false`.

U ovom kodu, ako `TryAdd` vrati `false` jer dodavanje nije uspjelo zato što stavka s tim ključem već postoji, Vi stavku svejedno dodajete preko indeksera. Name potrebe koristiti `TryAdd`. Umjesto toga, ja bih koristio `Add` koji će baciti iznimku ako stavka s tim ključem već postoji.